app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -44,6 +44,9 @@ def ready?
       vm_status == 'running'
     end
 
+    def reload
+    end
```

Comment:
@ShimShtein Introducing this method in our wrapper class, solves the error for us since it finds the method definition. But, I'm not very sure of what should be returned here?

Reply:
`#reload` [method](https://github.com/theforeman/foreman/blob/15fed2f6e10daf20010d3f292c71b8457a8edaf3/app/controllers/compute_resources_vms_controller.rb#L119) does need any return value.

Reply:
The vm is reloaded for getting the new power state to show in the UI. In our case, we load the power state from SDK each access to `vm_status` method, it can remain blank.